data/content_scripts/posting_process/context_menu.js

```diff
@@ -35,7 +35,7 @@ postingProcess.menuClickHandler = function(node, data) {
     text: selectionText,
     pageURL: window.location.href,
   };
-  self.postMessage(info);
+  self.postMessage({name: "startPostingProcess", content: info});
```

Comment:
This should use the context messenger but there are some problems while integrating this with the context messenger --
* Context Menus use `self.postMessage` instead of self.port.emit. Context menu scripts are in the same (message)context as `CONTENT_SCRIPT`, which means that I can't always send messages using `self.port.emit` if the context is `CONTENT_SCRIPT`.
This issue can be solved by setting a special variable before calling `Privly.message.messageExtension` and so in the message adapter, we can check for the variable and use `self.postMessage` instead of `self.port.emit`.

* Another issue is adding a listener. When using `self.postMessage`, an option is passed as `onMessage` during the initialization of the context menu as done [here](https://github.com/privly/privly-jetpack/blob/master/lib/posting_process.js#L91). I can't think of a way in which we can allow the adapter to set the listener.